packages/airnode-node/src/workers/local-gateways/validation.ts

```diff
@@ -102,6 +120,170 @@ export function verifyHttpSignedDataRequest(
+  airnodeAddress: z.string(),
+  endpointId: z.string(),
+  encodedParameters: z.string(),
+  encodedValue: z.string().optional(),
+  timestamp: z.string().optional(),
+  signature: z.string().optional(),
+});
+
+export const signOevDataBodySchema = z.object({
+  chainId: z.number().positive(),
+  dapiServerAddress: z.string(),
+  oevProxyAddress: z.string(),
+  updateId: z.string(),
+  bidderAddress: z.string(),
+  bidAmount: z.string(),
+  signedData: z.array(beacon),
+});
+
+type Beacon = z.infer<typeof beacon>;
+interface BeaconDecoded extends Required<Beacon> {
+  decodedValue: ethers.BigNumber;
+}
+
+export type ProcessSignOevDataRequestBody = z.infer<typeof signOevDataBodySchema>;
+
+export function dropInvalidBeacons(beacons: Beacon[]) {
+  const currentTime = new Date();
+
+  return beacons.reduce((result, beacon) => {
+    const goProcessing = goSync(() => {
+      const { airnodeAddress, endpointId, encodedParameters, encodedValue, timestamp, signature } = beacon;
+
+      // We don't have data for this beacon
+      if (!encodedValue || !timestamp || !signature) {
+        throw new Error('Missing beacon data');
+      }
+
+      // The timestamp is older than the last 2 minutes
+      if (isBefore(toDate(Number(timestamp)), addMinutes(currentTime, -TIMESTAMP_DEVIATION))) {
+        throw new Error('Beacon timestamp too old');
+      }
+
+      // To check parameters validity, exception is caught by the goSync
+      const _decodedParameters = decode(encodedParameters);
+      const decodedValue = ethers.BigNumber.from(ethers.utils.defaultAbiCoder.decode(['int256'], encodedValue)[0]);
+
+      const template: ApiCallTemplateWithoutId = {
+        airnodeAddress,
+        endpointId,
+        encodedParameters,
+      };
+      const templateId = getExpectedTemplateIdV1(template);
+
+      const message = ethers.utils.arrayify(
+        ethers.utils.keccak256(
+          ethers.utils.solidityPack(['bytes32', 'uint256', 'bytes'], [templateId, timestamp, encodedValue])
+        )
+      );
+      const addressFromSignature = ethers.utils.verifyMessage(message, signature);
+
+      // Signature doesn't match
+      if (airnodeAddress !== addressFromSignature) {
+        throw new Error('Invalid beacon signature');
+      }
+
+      if (decodedValue.gt(INT224_MAX) || decodedValue.lt(INT224_MIN)) {
+        throw new Error('Beacon value not within range');
+      }
+
+      return decodedValue;
+    });
+    if (!goProcessing.success) {
+      logger.log(`Invalid beacon data: ${goProcessing.error}`);
+      return result;
+    }
+
+    const decodedValue = goProcessing.data;
+    return [...result, { ...(beacon as Required<Beacon>), decodedValue }];
+  }, [] as BeaconDecoded[]);
+}
+
+function beaconCompare(a: BeaconDecoded, b: BeaconDecoded) {
+  if (a.decodedValue.lt(b.decodedValue)) return -1;
+  if (a.decodedValue.gt(b.decodedValue)) return 1;
+  return 0;
+}
+
+export function dropInconsistentBeacons(beacons: BeaconDecoded[]) {
```

Comment:
```suggestion
// TODO: Nit: rename to "chooseMaximumConsistentSet" or something
export function dropInconsistentBeacons(beacons: BeaconDecoded[]) {
  const sortedBeacons = [...beacons].sort(beaconCompare);
  let maxConsistent: BeaconDecoded[] = [];

  // Check every beacon interval. The maximum consistent set of beacons for aggregation must have every element
  // deviating from the average by less than 1%.
  for (let start = 0; start < sortedBeacons.length; start++) {
    const tmp = [sortedBeacons[start]];
    for (let end = start + 1; end < sortedBeacons.length; end++) {
      tmp.push(sortedBeacons[end]);
      // TODO: Extract to helper fn
      const avg = tmp.reduce((sum, beacon) => sum.add(beacon.decodedValue), ethers.BigNumber.from(0)).div(tmp.length);
      const maxDeviation = avg.div(100).mul(1);
      if (
        // TODO: Can be extracted as well
        avg.sub(tmp.at(0)!.decodedValue).abs().lte(maxDeviation) &&
        avg.sub(tmp.at(-1)!.decodedValue).abs().lte(maxDeviation)
      ) {
        if (tmp.length > maxConsistent.length) maxConsistent = tmp;
      }
    }
  }

  return maxConsistent;
}
```

Reply:
Based on https://api3workspace.slack.com/archives/C048B0DFMQR/p1677939910239879?thread_ts=1677744170.618349&cid=C048B0DFMQR

(The code above was not run or tested so test it as well).